packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart

```diff
@@ -26,35 +26,52 @@ import 'package:meta/meta.dart';
 @immutable
 class CognitoUserAttributeKey extends UserAttributeKey
     with AWSEquatable<CognitoUserAttributeKey>, AWSDebuggable {
-  const CognitoUserAttributeKey._(this._key, {this.readOnly = false})
-      : isCustom = false;
+  const CognitoUserAttributeKey._(
+    super.key, {
+    this.readOnly = false,
+    this.isCustom = false,
+  });
 
   /// Creates a custom Cognito attribute.
-  const CognitoUserAttributeKey.custom(this._key)
-      : isCustom = true,
-        readOnly = false;
+  factory CognitoUserAttributeKey.custom(String key) {
```

Comment:
Not a huge deal. I'd prefer this being `const` and using the `key` override like before, though.

Reply:
I was trying to simplify this, but yeah - maybe the small amount of extra complexity is worth it to make this a const constructor like the others.

Reply:
For sure, yeah it's always a balance